### PR TITLE
project level application

### PR DESCRIPTION
**Pull request by laojun**
## What type of PR is this?

- [x] Feature
- [ ] Bugfix
- [ ] Test
- [ ] Documentation
- [ ] Refactor
- [ ] Style
- [ ] Chore

## What this PR does / why we need it:
1、add isDeployingApp in application detail
2、hide deloy menu when application isDeployingApp
3、add application filter when application isDeployingApp

## Which issue(s) this PR fixes:

Fixes #

## Does this PR introduce a user interface change?

- [x] Yes(screenshot is required)
- [ ] No

## Special notes for your reviewer:

image

image

### CODE REVIEW

File: shell/app/modules/application/common/app-selector.tsx [hunk 1]
```diff
@@ -14,24 +14,53 @@
...
+    const appItem = existApp[item.value] || {};
+    return item.label ? item : { ...appItem, ...item, label: appItem.displayName || appItem.name };
+  });
+
+  const lackLabelItems = filter(reValue, item => !item.label);
+  if (isEmpty(lackLabelItems)) {
+    return reValue;
+  }
+
+  // 若还缺label，则从接口取
```
Comment (daskyrk) [inline shell/app/modules/application/common/app-selector.tsx:54]: Use english plz. If list is long, is there any performance issue?

File: shell/app/modules/application/types/application.d.ts [hunk 2]
```diff
@@ -108,6 +108,7 @@ interface IApplication {
   unBlockStart: string,
   unBlockEnd: string,
   blockStatus: PROJECT.BlockStatus;
+  isDeployingApp: boolean;
```
Comment (daskyrk) [inline shell/app/modules/application/types/application.d.ts:111]: This property name is strange. Sounds like a deploying flag.

File: shell/app/modules/application/pages/release/release-list.tsx [hunk 3]
```diff
@@ -99,22 +101,23 @@ const ReleaseList = () => {
...
+      <div className="release-list-page v-flex-box">
+        <IF check={appDetail.isDeployingApp}>
+          <AppSelector 
+            projectId={`${projectId}`} 
+            className='mb8 mx16'
+            allowClear
+            onChange={(_appId: number) => {
+              update({
+                applicationId: _appId,
+                queryObj: {
```
Comment (McDaddy) [inline shell/app/modules/application/pages/release/release-list.tsx:113]: how about set type of `queryObj ` as `null |  { q?: string, branchName?: string }`

File: shell/app/modules/application/common/app-selector.tsx [hunk 4]
```diff
@@ -14,24 +14,53 @@
...
+  onClickItem?: (arg: IApplication) => void;
+  onChange?: (arg: number) => void;
 }
 
 const AppItem = (app: IApplication) => {
   return <Tooltip key={app.id} title={app.name}>{app.displayName || app.name}</Tooltip>;
 };
 
+
+export const chosenItemConvert = (values: any) => {
```
Comment (McDaddy) [inline shell/app/modules/application/common/app-selector.tsx:39]: could this value type be specified？

File: shell/app/modules/application/common/app-selector.tsx [hunk 5]
```diff
@@ -14,24 +14,53 @@
...
+  const curApp = appStore.getState(s => s.detail);
+  const curValues = isArray(values) ? values : (values && [values]);
+  const existApp = {};
+  if(curApp.id) existApp[curApp.id] = curApp;
+  const reValue = map(curValues, item => {
+    const appItem = existApp[item.value] || {};
+    return item.label ? item : { ...appItem, ...item, label: appItem.displayName || appItem.name };
```
Comment (McDaddy) [inline shell/app/modules/application/common/app-selector.tsx:46]: if `appItem` is `{}`. label would be undefined, is this in line with expectations?

**Review by daskyrk — APPROVED** [review]

**Pull request merged**